server/netty-server/sync/src/main/scala/sttp/tapir/server/netty/sync/internal/ox/OxDispatcher.scala

```diff
@@ -20,12 +23,13 @@ import ox.channels.Actor
   */
 private[sync] class OxDispatcher()(using ox: Ox):
   private class Runner:
-    def runAsync(thunk: Ox ?=> Unit, onError: Throwable => Unit): Unit =
-      fork {
+    def runAsync(thunk: Ox ?=> Unit, onError: Throwable => Unit, cf: CompletableFuture[CancellableFork[Unit]]): Unit =
+      cf.complete(forkCancellable {
         try supervised(thunk)
-        catch case e => onError(e)
-      }.discard
+        catch case NonFatal(e) => onError(e)
```

Comment:
Changing to `NonFatal` to avoid situation when an exception calls `onError` which cancels the fork, which would trigger this catch again with `InterruptedException`.

Reply:
+1